src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5524,7 +5527,7 @@ private bool IsUsingAliasInScope(string name)
                         {
                             // Can't dot into the null literal or stackalloc expressions.
                             if ((boundLeft.Kind == BoundKind.Literal && ((BoundLiteral)boundLeft).ConstantValueOpt == ConstantValue.Null) ||
-                                boundLeft.Kind == BoundKind.StackAllocArrayCreation)
+                                boundLeft.Kind == BoundKind.StackAllocArrayCreation && MessageID.IDS_FeatureNestedStackalloc.RequiredVersion() > Compilation.LanguageVersion)
```

Comment:
>IDS_FeatureNestedStackalloc [](start = 97, length = 27)

Seems like we should be reporting a language version error in some cases here.

Reply:
I don't believe this is a realistic coding scenario (i.e. I don't believe a language version error here would benefit any programmers), but I'll do that.

---
In reply to: [207691295](https://github.com/dotnet/roslyn/pull/28969#discussion_r207691295) [](ancestors = 207691295)

Reply:
As you can see from `CannotDotIntoStackAllocExpression` in `StackAllocInitializerTests.cs`, there would be a language version error here on the stackalloc expression itself, so there doesn't need to be one on the member access expression. This test just ensures our error recovery is the same in C# 7.3, as the member access is permitted in C# 8.

---
In reply to: [208325668](https://github.com/dotnet/roslyn/pull/28969#discussion_r208325668) [](ancestors = 208325668,207691295)

Reply:
Further testing has revealed that this was dead code in C# 7.3 because of the error on the stackalloc expression. I'll just remove the special case for stackalloc here.

---
In reply to: [208329378](https://github.com/dotnet/roslyn/pull/28969#discussion_r208329378) [](ancestors = 208329378,208325668,207691295)